### ISSUE TITLE
508-defect-2 [FOCUS MANAGEMENT, SCREENREADER]: Upon choice to Edit permanent address or Add a temporary address, focus MUST be managed and context change communicated

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9941]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On step 1 of 3 Veteran information, Shipping address, if a user selects to Edit permanent address or Add a temporary address, where focus is and that there is a context change **must** be communicated.

When a user selects to edit or add an address, the context of the form changes. At present, the screen reader goes silent and there is nothing communicated that the screen has changed. Once the user triggers the screen reader to start reading again, the focus is on the h3 Shipping address. This flow is unclear for a non-sighted user.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to know if my focus is moved so that I understand there is a context change.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/veteran-information/addresses in browser, logged in as 228
2. Start screenreading device listed in Environment
3. Navigate to Permanent address
4. Select 'Edit permanent address' link
5. Verify that the context change is not communicated
6. Verify that when the screen reader starts reading again, focus is on Shipping address, but without communication that it is now a new form

## Possible Fixes (optional)

When a user selects 'Edit permanent address' or 'Add a temporary address', the edit in place functionality should be revealed, focus should be on the "Edit permanent address" or "Add a temporary address" heading.

## WCAG or Vendor Guidance (optional)

* [wuhcag on input](https://www.wuhcag.com/on-input/)

* [W3C WCAG On Input: Understanding SC 3.2.2 On Input](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html)

**Comment by mr0sari0** [comment]
Hey @jenstrickland, I am testing on Chrome with ChromeVox. On user 1, when I click the edit permanent address link, I hear "edit permanent address button". The same with the add a temporary address link.

**Comment by mr0sari0** [comment]
Ah, nevermind, I just read your message about the bug being on user 228.

**Comment by mr0sari0** [comment]
Looks like there's no address data being displayed on staging with this user. Once we get the data sorted out, it should read the edit address message from my first comment.

**Comment by jenstrickland** [comment]
When a user clicks the button, focus should move to the now open edit in place functionality. We need to communicate the context change to the screen reader. I can show you an example in the authenticated experience flow. 

Also, I noticed you said you're using Chrome and ChromeVox. Are you on Windows or Mac?

**Comment by mr0sari0** [comment]
Hi @jenstrickland, could you please review this again?

**Comment by jenstrickland** [comment]
@mr0sari0 — I just tested this in staging, but focus does not seem to be managed. When I select Edit this address in the Permanent address section, focus moved to the label for the checkbox for the military base addresses, and when I saved the address change, focus moved to the page breadcrumbs.

Are the changes available in staging, yet?

**Comment by mr0sari0** [comment]
Hi @jenstrickland, the changes should be in staging now.

**Comment by jenstrickland** [comment]
Validated. 🎉
